test/sql/storage/catalog/test_table_macro_storage.test

```diff
@@ -0,0 +1,78 @@
+# name: test/sql/storage/catalog/test_table_macro_storage.test
+# description: Create and drop a table macro over different runs
+# group: [catalog]
+
+# load the DB from disk
+load __TEST_DIR__/macro_storage.db
+
+statement ok
+PRAGMA wal_autocheckpoint='1TB';
+
+statement ok
+CREATE TABLE test_tbl (id INT, name string, height double);
+
+statement ok
+INSERT INTO  test_tbl values (1,'tom', 1.1), (2,'dick',1.2),(3,'harry', 1.2),
+                             (4,'mary',0.9), (5,'mungo', 0.8), (6,'midge', 0.5);
+
+
+# create a table macro
+statement ok
+CREATE MACRO xt(a, _name) as TABLE SELECT * FROM test_tbl WHERE id<=a or name = _name;
+
+
+# use the macro
+query III
+SELECT * FROM xt(10, '*') ORDER BY  height limit 1;
+----
+6	midge	0.5   
+
+
+statement ok
+DROP MACRO TABLE xt;
```

Comment:
The reason the storage is not correctly tested is because you are dropping the macro before the restart - if you remove this drop macro statement here then it should be correctly tested.

Reply:
Have redone this script 
Now checks regular storage of table macros and WAL storage